docs/TIPS.md

```diff
@@ -0,0 +1,251 @@
+              pyston fannkuch.py                        :    6.1s baseline: 5.8 (+5.6%)
+              pyston chaos.py                           :   18.2s baseline: 17.3 (+5.2%)
+              pyston fasta.py                           :    4.4s baseline: 4.1 (+7.8%)
+              pyston pidigits.py                        :    5.5s baseline: 5.4 (+2.5%)
+              pyston richards.py                        :    1.5s baseline: 1.5 (+0.6%)
+              pyston deltablue.py                       :    1.4s baseline: 1.3 (+1.0%)
+              pyston (geomean-10eb)                     :    4.3s baseline: 4.2 (+2.7%)
+```
+
+The numbers shown is the smallest value obtained after 3 (or more, if you wish) runs. It is recommended to include these outputs in non-trivial pull requests as a way to document the performance impacts, if any.
+
+##### Problem: My trivial change just created a 2% reduction in performance
+
+Unfortunately, variations in performance can be fairly large across runs, relatively speaking. The variance usually isn't more than 1-2%, but given the breadth of Python code, performance is can sometimes only be achieved via dozens of 0.5% optimizations and it is hard to measure a 0.5% performance change if the variance is 1%. So depending on the machine Pyston is running on, the variance can feel quite large. For small optimizations, it's useful to write a small microbenchmark (e.g. a tight loop).
+
+To add to the challenge of precise performance measurements, *adding trivial code or even unreachable code can change the performance characteristics* by moving other hot functions around in the assembly which can help or hinder the effectiveness of the instruction cache. This is something we plan to address in the future, but reducing variance across runs would be a very valuable contribution if you are interested (e.g. by placing large gaps between assemblies of different functions and aligning them to cache lines).
+
+Some benchmarks are particularly prone to variations caused by trivial changes. [http://speed.pyston.org/](http://speed.pyston.org/timeline) shows the performance graphs of many of the benchmarks. If a benchmark already shows a fair amount of up and downs, it is not as big a deal if the PR causes a performance drop in that benchmark. For example, [django_template](http://speed.pyston.org/timeline/#/?exe=1,3&base=none&ben=django_template&env=2&revs=1000&equid=off) is a benchmark that performs better and better, but [nbody](http://speed.pyston.org/timeline/#/?exe=1,3&base=none&ben=nbody&env=2&revs=1000&equid=off) has the tendency to be affected by the most unrelated changes.
+
+### Profiling tools
+
+The general go-to tool for profiling that we use is `perf`, which shows the hot functions in which the program spends the most time and their callers. It can be used as follows:
+
+```
+perf record -g -- ./pyston_release ../pyston-perf/benchmarking/benchmark_suite/django_template.py
```

Comment:
I'd put `-p -q` here in this command line.  it's so useful that I don't think we want to suggest people should do a perf run without it.